include/deal.II/base/derivative_form.h

```diff
@@ -63,9 +63,9 @@ class DerivativeForm
   DerivativeForm ();
 
   /**
-   * Constructor from a second order tensor.
+   * Constructor from a tensor.
    */
-  DerivativeForm (const Tensor<2,dim,Number> &);
+  DerivativeForm (const Tensor<order+1,dim,Number> &);
```

Comment:
Nice improvement in itself. Can you add an entry to changes.h for just this?

In the future, feel free to isolate these changes to a single file and submit them as separate pull requests.